src/cpp/handler/controlrequest.cpp

```diff
@@ -33,15 +33,15 @@ namespace ControlRequest {
 class ConnCheck : public Deferred
 {
 	Q_OBJECT
-
+	
 	Connection finishedConnection;
 
 public:
-	ConnCheck(ZrpcManager *controlClient, const CidSet &cids, QObject *parent = 0) :
-		Deferred(parent)
+	ConnCheck(ZrpcManager *controlClient, const CidSet &cids) :
+		Deferred()
 	{
-		ZrpcRequest *req = new ZrpcRequest(controlClient, this);
-		finishedConnection = req->finished.connect(boost::bind(&ConnCheck::req_finished, this, req));
+		req = std::make_unique<ZrpcRequest>(controlClient, this);
```

Comment:
Passing `this` as the parent should be removed here, since the lifetime is already managed by unique_ptr and there is a risk of double-deletion if QObject parenting is used at the same time.

Reply:
zrprequest is still not updated in this regard yet. 
```
ZrpcRequest(ZrpcManager *manager, QObject *parent = 0);
```